tests/unit/test_svs.cpp

```diff
@@ -0,0 +1,2069 @@
+        /* SVS-Vamana specifics */
+        .alpha = 0.9,
+        .graph_max_degree = 64,
+        .construction_window_size = 20,
+        .max_candidate_pool_size = 1024,
+        .prune_to = 60,
+        .use_search_history = VecSimOption_ENABLE,
+    };
+
+    VecSimIndex *index = this->CreateNewIndex(params);
+
+    // To meet accurary in LVQ case we have to add bulk of vectors at once.
+    std::vector<TEST_DATA_T[dim]> v(n);
+    std::vector<size_t> ids(n);
+
+    for (size_t i = 0; i < n; i++) {
+        auto &f = v[i];
+        f[0] = TEST_DATA_T(i + 1) / n;
+        for (size_t j = 1; j < dim; j++) {
+            f[j] = 1.0;
+        }
+        // Use as label := n - (internal id)
+        ids[i] = n - i;
+    }
+
+    auto svs_index = this->CastToSVS(index);
+    svs_index->addVectors(v.data(), ids.data(), n);
+
+    ASSERT_EQ(VecSimIndex_IndexSize(index), n);
+    TEST_DATA_T query[dim];
+    query[0] = 1.1;
+    for (size_t i = 1; i < dim; i++) {
+        query[i] = 1.0;
+    }
+    auto verify_res = [&](size_t id, double score, size_t result_rank) {
+        ASSERT_EQ(id, result_rank + 1);
+        TEST_DATA_T expected_score = index->getDistanceFrom_Unsafe(id, query);
+        // Verify that abs difference between the actual and expected score is at most 1/10^5.
+        ASSERT_NEAR((TEST_DATA_T)score, expected_score, 1e-5f);
+    };
+
+    uint expected_num_results = 31;
+    // Calculate the score of the 31st distant vector from the query vector (whose id should be 30)
+    // to get the radius.
+    VecSim_Normalize(query, dim, params.type);
+    double radius = index->getDistanceFrom_Unsafe(31, query);
+    runRangeQueryTest(index, query, radius, verify_res, expected_num_results, BY_SCORE);
+    // Return results BY_ID should give the same results.
+    runRangeQueryTest(index, query, radius, verify_res, expected_num_results, BY_ID);
+
+    VecSimIndex_Free(index);
+}
+
+TYPED_TEST(SVSTest, FitMemoryTest) {
+    size_t dim = 4;
+    SVSParams params = {
+        .dim = dim,
+        .metric = VecSimMetric_L2,
+        .blockSize = DEFAULT_BLOCK_SIZE,
+        /* SVS-Vamana specifics */
+        .alpha = 1.2,
+        .graph_max_degree = 64,
+        .construction_window_size = 20,
+        .max_candidate_pool_size = 1024,
+        .prune_to = 60,
+        .use_search_history = VecSimOption_ENABLE,
+    };
+
+    VecSimIndex *index = this->CreateNewIndex(params);
+
+    size_t initial_memory = index->getAllocationSize();
+    index->fitMemory();
+    ASSERT_GE(index->getAllocationSize(), initial_memory);
+
+    // Add vector
+    GenerateAndAddVector<TEST_DATA_T>(index, dim, 0);
+    initial_memory = index->getAllocationSize();
+    index->fitMemory();
+    size_t final_size = index->getAllocationSize();
+    // Due to the initial capacity, the memory for the vector was already allocated
+    ASSERT_EQ(final_size, initial_memory);
+
+    VecSimIndex_Free(index);
+}
+
+TYPED_TEST(SVSTest, resolve_ws_search_runtime_params) {
```

Comment:
@alonre24 , please check if tests here and below meet this request: https://github.com/RedisAI/VectorSimilarity/pull/598#discussion_r1986315445